serviceregistry-server/src/test/java/no/difi/meldingsutveksling/serviceregistry/controller/ServiceRecordControllerTest.java

```diff
@@ -0,0 +1,173 @@
+        when(entityService.getEntityInfo("43")).thenReturn(ASinfo);
+        CitizenInfo citizenInfo = new CitizenInfo("12345678901");
+        when(entityService.getEntityInfo("12345678901")).thenReturn(citizenInfo);
+
+        EDUServiceRecord dpoServiceRecord = new EDUServiceRecord(null, "pem123", "http://foo", "123", "321", "42");
+        when(serviceRecordFactory.createEduServiceRecord("42")).thenReturn(dpoServiceRecord);
+
+        ServiceregistryProperties serviceregistryProperties = new ServiceregistryProperties();
+        ServiceregistryProperties.PostVirksomhet postVirksomhet = new ServiceregistryProperties.PostVirksomhet();
+        postVirksomhet.setEndpointURL(new URL("http://foo"));
+        serviceregistryProperties.setDpv(postVirksomhet);
+        PostVirksomhetServiceRecord dpvServiceRecord = new PostVirksomhetServiceRecord(serviceregistryProperties, "43");
+        when(serviceRecordFactory.createPostVirksomhetServiceRecord("43")).thenReturn(dpvServiceRecord);
+
+        KontaktInfo kontaktInfoMock = mock(KontaktInfo.class);
+        when(kontaktInfoMock.getCertificate()).thenReturn("cert123");
+        when(kontaktInfoMock.getOrgnrPostkasse()).thenReturn("post123");
+        when(kontaktInfoMock.getPostkasseAdresse()).thenReturn("adr123");
+        when(kontaktInfoMock.isNotifiable()).thenReturn(false);
+        when(kontaktInfoMock.getEpostadresse()).thenReturn("post@post.foo");
+        when(kontaktInfoMock.getMobiltelefonnummer()).thenReturn("123456789");
+        when(kontaktInfoMock.isReservert()).thenReturn(false);
+        PostAddress postAddress = mock(PostAddress.class);
+        PostAddress returnAddress = mock(PostAddress.class);
+        SikkerDigitalPostServiceRecord dpiServiceRecord = new SikkerDigitalPostServiceRecord(null, kontaktInfoMock, ServiceIdentifier.DPI, "12345678901", postAddress, returnAddress);
+        when(serviceRecordFactory.createServiceRecordForCititzen(eq("12345678901"), any(), any())).thenReturn
+                (dpiServiceRecord);
+    }
+
+    @Test
+    public void entityShouldReturnOK() throws Exception {
+        mvc.perform(get("/identifier/42").accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.serviceRecord.organisationNumber", is("42")))
+                .andExpect(jsonPath("$.serviceRecord.serviceIdentifier", is("EDU")))
+                .andExpect(jsonPath("$.serviceRecord.pemCertificate", is("pem123")))
+                .andExpect(jsonPath("$.serviceRecord.serviceCode", is("123")))
+                .andExpect(jsonPath("$.serviceRecord.serviceEditionCode", is("321")))
+                .andExpect(jsonPath("$.serviceRecord.endPointURL", is("http://foo")))
+                .andExpect(jsonPath("$.infoRecord.identifier", is("42")))
+                .andExpect(jsonPath("$.infoRecord.entityType.name", is("Organisasjonsledd")))
+                .andExpect(jsonPath("$.infoRecord.entityType.acronym", is("ORGL")));
```

Comment:
Kanskje ta bort noen av expect-ene og kanskje generalisert noen av jsonPath-ene så det blir mindre sjanse for å bryte testen bare man gjør en mindre refaktorering? I hvertfall kanskje splitta expect status ok og EDU i en separat test.